docs/environments.md

```diff
@@ -0,0 +1,82 @@
+# Environments
+
+Environments is a feature that allows you to deploy the same project to multiple places.
+
+## Backwards compatibilty
+
+Legacy `wrangler.toml` files will still work as expected during the initial rollout of this feature, however you may notice a deprecation warning. One of the goals of environments is to make it more obvious when you are deploying to a traditional worker with routes, and when you are deploying to a subdomain on workers.dev.
+
+A `wrangler.toml` before this release looks like this:
+
+```toml
+name = "my-worker"
+type = "webpack"
+account_id = "12345678901234567890"
+zone_id = "09876543210987654321"
+private = false
+route = "example.com/*
+```
+
+With this configuration, Wrangler will behave in the following manner:
+
+`wrangler publish` will publish your worker to your subdomain on workers.dev
+`wrangler publish --release` will publish your worker to your route at `example.com/*`. 
+
+It is important to note that both of these commands will issue a deprecation warning. To remove these warnings, you can configure Wrangler with environments to separate deploys to workers.dev from deploys to workers routes.
+
+## Intended Use
+
+A common use case that this feature enables is deploying your worker to a staging subdomain before pushing your worker to production. The top-level configuration will be used when running `wrangler publish`. In addition, you can specify multiple environments that you would like to deploy code to.
+
+### Example
+
+```toml
+type = "webpack"
+name = "my-worker"
+account_id = "12345678901234567890"
+zone_id = "09876543210987654321"
+route = "example.com/*"
+
+[env.staging]
+name = "my-worker-staging"
+account_id = "12345678901234567890"
+zone_id = "09876543210987654321"
+route = "staging.example.com/*"
+
+[env.dev]
+name = "my-worker-dev"
+account_id = "12345678901234567890"
+zone_id = "09876543210987654321"
+route = "dev.example.com/*"
+```
```

Comment:
nit: if it was me (and many other very cautious devs out there) i'd likely have prod as a non-default env and dev as my default, making me less likely to fat-finger a production deploy. I'd also recommend this approach to others, even though in reality you could configure this any darn way you please.

also i suppose we should make it clear that you could configure this any darn way you please.